scripts/machine_translation/train_transformer.py

```diff
@@ -444,51 +282,51 @@ def train():
+                                                   batchify_fn=train_batchify_fn,
+                                                   num_workers=8)
+
+    val_batch_sampler = nlp.data.FixedBucketSampler(lengths=target_val_lengths,
+                                                    batch_size=args.test_batch_size,
+                                                    num_buckets=args.num_buckets,
+                                                    ratio=args.bucket_ratio,
+                                                    shuffle=False,
+                                                    use_average_length=True,
+                                                    bucket_scheme=bucket_scheme)
     logging.info('Valid Batch Sampler:\n{}'.format(val_batch_sampler.stats()))
-    val_data_loader = DataLoader(data_val,
-                                 batch_sampler=val_batch_sampler,
-                                 batchify_fn=test_batchify_fn,
-                                 num_workers=8)
-    test_batch_sampler = FixedBucketSampler(lengths=target_test_lengths,
-                                            batch_size=args.test_batch_size,
-                                            num_buckets=args.num_buckets,
-                                            ratio=args.bucket_ratio,
-                                            shuffle=False,
-                                            use_average_length=True,
-                                            bucket_scheme=bucket_scheme)
+    val_data_loader = gluon.data.DataLoader(data_val,
+                                            batch_sampler=val_batch_sampler,
+                                            batchify_fn=test_batchify_fn,
+                                            num_workers=8)
+    test_batch_sampler = nlp.data.FixedBucketSampler(lengths=target_test_lengths,
+                                                     batch_size=args.test_batch_size,
+                                                     num_buckets=args.num_buckets,
+                                                     ratio=args.bucket_ratio,
+                                                     shuffle=False,
+                                                     use_average_length=True,
+                                                     bucket_scheme=bucket_scheme)
     logging.info('Test Batch Sampler:\n{}'.format(test_batch_sampler.stats()))
-    test_data_loader = DataLoader(data_test,
-                                  batch_sampler=test_batch_sampler,
-                                  batchify_fn=test_batchify_fn,
-                                  num_workers=8)
+    test_data_loader = gluon.data.DataLoader(data_test,
+                                             batch_sampler=test_batch_sampler,
+                                             batchify_fn=test_batchify_fn,
+                                             num_workers=8)
```

Comment:
I think we can also put the code for sampler and dataloader into a separate file, including a function called make_dataloader, which returns train_data_loader, val_data_loader, and test_data_loader.